### PR TITLE
feature/GH-13918: expose function isGuestCart() in ActiveCartFacade

### LINKED ISSUE #13918: expose function `isGuestCart()` in `ActiveCartFacade`
Facade only can expose functions which return `void` or `observable`. We will make function `isGuestCart` returns `Observable<boolean>`, so that this function can be exposed.

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #13918

### CODE REVIEW

File: feature-libs/cart/main/core/facade/active-cart.service.ts [hunk 1]
```diff
@@ -474,21 +477,31 @@ export class ActiveCartService implements OnDestroy {
...
+  isGuestCart(cart?: Cart): Observable<boolean> {
+    if (cart) {
+      const cartUser = cart?.user;
+      return of(
+        Boolean(
+          cartUser &&
+            (cartUser.name === OCC_USER_ID_GUEST ||
+              this.isEmail(cartUser.uid?.split('|').slice(1).join('|')))
+        )
+      );
```
Comment (plabadie) [inline feature-libs/cart/main/core/facade/active-cart.service.ts]: These lines are repeated twice within `isGuestCart()`, is there a way we can avoid this duplication?

Comment (WeizhengSap) [reply feature-libs/cart/main/core/facade/active-cart.service.ts]: we can create a private function

File: feature-libs/checkout/components/guards/checkout-auth.guard.ts [hunk 2]
```diff
@@ -51,10 +51,7 @@ export class CheckoutAuthGuard implements CanActivate {
     );
   }
 
-  protected handleAnonymousUser(cartUser?: User): boolean | UrlTree {
-    if (this.activeCartService.isGuestCart()) {
-      return !!cartUser;
-    }
```
Comment (plabadie) [inline feature-libs/checkout/components/guards/checkout-auth.guard.ts:57]: Why we remove this?

Comment (WeizhengSap) [reply feature-libs/checkout/components/guards/checkout-auth.guard.ts:57]: we don't get `this.activeCartService.getAssignedUser()`, but get `this.activeCartFacade.isGuestCart()` directly. So no need to check `this.activeCartService.isGuestCart()` again.

**Review by plabadie — APPROVED** [review]

**Pull request merged**